### PR TITLE
ConfigObject: fix creation of new config file

### PR DESCRIPTION
**Pull request by Be-ing**
Fixing a regression from #3192

### CODE REVIEW

File: src/preferences/configobject.cpp [hunk 1]
```diff
@@ -237,10 +237,12 @@ bool ConfigObject<ValueType>::save() {
     }
 
     QFile oldConfig(m_filename);
-    if (!oldConfig.remove()) {
-        qWarning().nospace() << "Could not remove old config file: "
-                             << oldConfig.fileName() << ": " << oldConfig.errorString();
-        return false;
+    if (oldConfig.exists()) {
```
Comment (uklotzde) [inline src/preferences/configobject.cpp:240]: Combine into a single if statement

Comment (Be-ing) [reply src/preferences/configobject.cpp:240]: I'd prefer to keep them separate. QDir::remove is not just checking something, it has side effects. I think it's easier to understand that by separating these.

Comment (uklotzde) [reply src/preferences/configobject.cpp:240]: QFile::exists() also needs to access the file system. I don't see a benefit, short-circuit evaluation is an obvious choice here. 

Nevertheless we should a comment that removal of non-existent files will fail, no matter if we use two or a single if statement.

**Review by uklotzde — APPROVED** [review]
Thank you for stepping in! LGTM

**Comment by poelzi** [comment]
LGTM

**Pull request merged**